### ISSUE TITLE
Error: Cannot find module '@ronomon/utimes'

### ISSUE THREAD

**Issue by ElanX** [ChuckMac/tadpoles-scraper#5]
Anyone know how to fix this error?:
> Error: Cannot find module '@ronomon/utimes'

**Comment by ChuckMac** [comment]
have you tried:

`npm install @ronomon/utimes`

**Comment by ElanX** [comment]
No, thanks for the quick reply. I just tried it, and this was the output:

```
> @ronomon/utimes@2.3.1 install C:\Program Files\tadpoles-scraper\node_modules\@ronomon\utimes
> node-gyp rebuild


C:\Program Files\tadpoles-scraper\node_modules\@ronomon\utimes>if not defined npm_config_node_gyp (node "C:\Program Files\nodejs\node_modules\npm\node_modules\npm-lifecycle\node-gyp-bin\\..\..\node_modules\node-gyp\bin\node-gyp.js" rebuild )  else (node "C:\Program Files\nodejs\node_modules\npm\node_modules\node-gyp\bin\node-gyp.js" rebuild )
Warning: Missing input files:
C:\Program Files\tadpoles-scraper\node_modules\@ronomon\utimes\build\Release\binding.node
Building the projects in this solution one at a time. To enable parallel build, please add the "/m" switch.
  binding.cc
  win_delay_load_hook.cc
c:\program files\tadpoles-scraper\node_modules\@ronomon\utimes\binding.cc(151): error C2660: 'v8::Value::Uint32Value':
function does not take 0 arguments [C:\Program Files\tadpoles-scraper\node_modules\@ronomon\utimes\build\binding.vcxpro
j]
  c:\users\elan\appdata\local\node-gyp\cache\12.10.0\include\node\v8.h(2580): note: see declaration of 'v8::Value::Uint
  32Value' (compiling source file ..\binding.cc)
c:\program files\tadpoles-scraper\node_modules\@ronomon\utimes\binding.cc(151): error C2789: 'flags': an object of cons
t-qualified type must be initialized [C:\Program Files\tadpoles-scraper\node_modules\@ronomon\utimes\build\binding.vcxp
roj]
  c:\program files\tadpoles-scraper\node_modules\@ronomon\utimes\binding.cc(151): note: see declaration of 'flags'
c:\program files\tadpoles-scraper\node_modules\@ronomon\utimes\binding.cc(152): error C2660: 'v8::Value::IntegerValue':
 function does not take 0 arguments [C:\Program Files\tadpoles-scraper\node_modules\@ronomon\utimes\build\binding.vcxpr
oj]
  c:\users\elan\appdata\local\node-gyp\cache\12.10.0\include\node\v8.h(2578): note: see declaration of 'v8::Value::Inte
  gerValue' (compiling source file ..\binding.cc)
c:\program files\tadpoles-scraper\node_modules\@ronomon\utimes\binding.cc(152): error C2789: 'btime': an object of cons
t-qualified type must be initialized [C:\Program Files\tadpoles-scraper\node_modules\@ronomon\utimes\build\binding.vcxp
roj]
  c:\program files\tadpoles-scraper\node_modules\@ronomon\utimes\binding.cc(152): note: see declaration of 'btime'
c:\program files\tadpoles-scraper\node_modules\@ronomon\utimes\binding.cc(153): error C2660: 'v8::Value::IntegerValue':
 function does not take 0 arguments [C:\Program Files\tadpoles-scraper\node_modules\@ronomon\utimes\build\binding.vcxpr
oj]
  c:\users\elan\appdata\local\node-gyp\cache\12.10.0\include\node\v8.h(2578): note: see declaration of 'v8::Value::Inte
  gerValue' (compiling source file ..\binding.cc)
c:\program files\tadpoles-scraper\node_modules\@ronomon\utimes\binding.cc(153): error C2789: 'mtime': an object of cons
t-qualified type must be initialized [C:\Program Files\tadpoles-scraper\node_modules\@ronomon\utimes\build\binding.vcxp
roj]
  c:\program files\tadpoles-scraper\node_modules\@ronomon\utimes\binding.cc(153): note: see declaration of 'mtime'
c:\program files\tadpoles-scraper\node_modules\@ronomon\utimes\binding.cc(154): error C2660: 'v8::Value::IntegerValue':
 function does not take 0 arguments [C:\Program Files\tadpoles-scraper\node_modules\@ronomon\utimes\build\binding.vcxpr
oj]
  c:\users\elan\appdata\local\node-gyp\cache\12.10.0\include\node\v8.h(2578): note: see declaration of 'v8::Value::Inte
  gerValue' (compiling source file ..\binding.cc)
c:\program files\tadpoles-scraper\node_modules\@ronomon\utimes\binding.cc(154): error C2789: 'atime': an object of cons
t-qualified type must be initialized [C:\Program Files\tadpoles-scraper\node_modules\@ronomon\utimes\build\binding.vcxp
roj]
  c:\program files\tadpoles-scraper\node_modules\@ronomon\utimes\binding.cc(154): note: see declaration of 'atime'
gyp ERR! build error
gyp ERR! stack Error: `C:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools\MSBuild\15.0\Bin\MSBuild.exe` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onExit (C:\Program Files\nodejs\node_modules\npm\node_modules\node-gyp\lib\build.js:196:23)
gyp ERR! stack     at ChildProcess.emit (events.js:209:13)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:272:12)
gyp ERR! System Windows_NT 10.0.17763
gyp ERR! command "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild"
gyp ERR! cwd C:\Program Files\tadpoles-scraper\node_modules\@ronomon\utimes
gyp ERR! node -v v12.10.0
gyp ERR! node-gyp -v v5.0.3
gyp ERR! not ok
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @ronomon/utimes@2.3.1 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the @ronomon/utimes@2.3.1 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\elan\AppData\Roaming\npm-cache\_logs\2019-09-05T19_40_48_630Z-debug.log
```

**Comment by ChuckMac** [comment]
Looks like you are on Windows... do you have the npm Windows build tools installed?

`npm install --global windows-build-tools`

**Comment by ElanX** [comment]
> Looks like you are on Windows... do you have the npm Windows build tools installed?
> 
> `npm install --global windows-build-tools`

Yes, Windows. Windows build tools installed with no problems. The first error I got was mkdirp was missing, but that installed fine. Which is where I'm up to, trying to install @ronomon/utimes

I realize this isn't any issue with your scraper, but let me know if you have any ideas to get me going.

**Comment by ChuckMac** [comment]
Yea Windows build failures for other packages aren't really my thing... only other thought offhand is do you have Python 2.7.x installed and is that what node is configured to use?

`npm config set python python2.7`

**Comment by ElanX** [comment]
> Yea Windows build failures for other packages aren't really my thing... only other thought offhand is do you have Python 2.7.x installed and is that what node is configured to use?
> 
> `npm config set python python2.7`

Yes, Python 2.7.16 and ran npm config set python python2.7

I see my **npm install @ronomon/utimes** installed node_modules into my tadpole-scraper directory. Where should I be running that install so my nodes_modules are in the correct place and in path?

**Comment by BernhardEgger** [comment]
Hi ElanX,

I have the exact same issue here.
Did you figure it out at the end?

Best
Bernhard

**Comment by BernhardEgger** [comment]
Hi,

seems to be an issue related to Node.js 12
https://github.com/ronomon/base64/issues/1

I installed Node.js 10 here:
https://nodejs.org/en/download/releases/

That fixed it for me

Perhaps you can add a note in the Readme?

Thank you so much for putting this tool together - saved me a lot of time :)

Best
Bernhard

**Comment by Sheridan-Tech** [comment]
I too have this issue.

**Comment by Sheridan-Tech** [comment]
[SOLVED] I uninstalled node.js 12 in favor of 10. Then after a bit of work I was able to start downloading! So exciting. It was taking forever to download the files individually. They should really give you a batch download function.

**Comment by ChuckMac** [comment]
fixed with #6
